compiler-lib/src/strtab.rs

```diff
@@ -37,61 +64,61 @@ mod tests {
     use super::*;
     use std::collections::HashMap;
 
-    macro_rules! get {
-        ($tab: expr) => {
-            unsafe { &mut *$tab.entries.get() }
-        };
-    }
-
     #[test]
     fn no_duplication() {
-        let strtab = StringTable::new();
-
-        strtab.intern("foo");
-        strtab.intern("foo");
-        strtab.intern("foo");
-        assert_eq!(1, get!(strtab).len());
-
-        strtab.intern("bar");
-        strtab.intern("bar");
-        strtab.intern("foo");
-        assert_eq!(2, get!(strtab).len());
+        let mut strtab = StringTable::new();
+
+        let a = strtab.intern("foo");
+        let b = strtab.intern("foo");
+        let c = strtab.intern("foo");
+        assert_eq!(1, strtab.entries.len());
+        assert_eq!(a, b);
+        assert_eq!(a, c);
```

Comment:
an explicit cast to pointers and pointer comparison would be good

Reply:
That's what eq does